src/qc/bit_vector.rs

```diff
@@ -106,83 +47,85 @@ where
     // TODO: later with SNARKs we'll use a smaller verifier parameter
     type QCVerifierParams = QCParams<A::VerificationKey, A::PublicParameter>;
 
-    type Proof = BitVec;
+    type QC = (A::Signature, BitVec);
     type MessageLength = U32;
-    type CheckedType = ();
+    type QuorumSize = U256;
 
-    fn partial_sign<R: CryptoRng + RngCore>(
+    fn sign<R: CryptoRng + RngCore>(
         agg_sig_pp: &A::PublicParameter,
         message: &GenericArray<A::MessageUnit, Self::MessageLength>,
-        sig_key: &A::SigningKey,
+        sk: &A::SigningKey,
         prng: &mut R,
     ) -> Result<A::Signature, PrimitivesError> {
-        A::sign(agg_sig_pp, sig_key, message, prng)
+        A::sign(agg_sig_pp, sk, message, prng)
     }
 
     fn assemble(
         qc_pp: &Self::QCProverParams,
-        active_keys: &BitSlice,
-        partial_sigs: &[A::Signature],
-    ) -> Result<(A::Signature, Self::Proof), PrimitivesError> {
-        if active_keys.len() != qc_pp.stake_entries.len() {
+        signers: &BitSlice,
+        sigs: &[A::Signature],
+    ) -> Result<Self::QC, PrimitivesError> {
+        if signers.len() != qc_pp.stake_entries.len() {
             return Err(ParameterError(format!(
                 "bit vector len {} != the number of stake entries {}",
-                active_keys.len(),
+                signers.len(),
```

Comment:
So this version is "each key gets one unit of stake". Is this something that is non-negotiable for this aggregation scheme? I'm worried that we might have a near-miss solution for what HotShot actually needs here.